### DISCUSSION TITLE
SMT Relay fails with "No SMTP servers were available for gmail.com. No hosts to try."

### DISCUSSION THREAD

**Discussion by jordantrizz** [postalserver/postal#2986] [Help with using Postal]
I'm trying to use an SMTP relay that is a Postfix server on the same host using 127.0.0.1 on port 2525. 

I can send using the sendmail command on the local instance, and via telnet outside and inside the docker container postal-smtp-1. Unfortunately, the docker container doesn't have sendmail or telnet but has the nc command which works.

I can't use dig, host or nslookup to see if it's a DNS resolution issue. I can see the /etc/resolv.conf is correct.

Postal works without using an SMTP relay, but the minute I configure an SMTP relay email fails with the error

```
No SMTP servers were available for gmail.com. No hosts to try.

Support Ref: KDYPAIWU
```

Here is my config

```
postal:
  web_hostname: postal.domain.com
  web_protocol: https
  smtp_hostname: postal.domian.com
  smtp_relays: [ smtp://127.0.0.1:2525 ]
```

It seems like this code is related, a DNS query is being done and then if it fails would spit out the same error.

```
    before do
      allow(DNSResolver.local).to receive(:a).with("mx1.example.com").and_return(dns_result)
      sender.start
    end

    context "when there is no current endpoint to use" do
      it "returns a SoftFail" do
        result = sender.send_message(message)
        expect(result).to be_a SendResult
        expect(result).to have_attributes(
          type: "SoftFail",
          retry: true,
          output: "",
          details: /No SMTP servers were available for example.com. No hosts to try./,
          connect_error: true
        )
      end
    end
```

**Comment by willpower232** [comment]
Are you specifying that relay via yaml or env? I don't think you have the right config

**Reply by jordantrizz** [reply]
Yaml as per this https://github.com/postalserver/postal/blob/main/doc/config/yaml.yml

**Reply by willpower232** [reply]
Ah of course, too many variants around. It would be worth looking for that support ref in your Postal SMTP logs to see what it did. 

There will probably be some clues left out from this file https://github.com/postalserver/postal/blob/d5104991903d42563c6c6f7ff01276d496e3304f/app/senders/smtp_sender.rb#L47

**Reply by jordantrizz** [reply]
> It would be worth looking for that support ref in your Postal SMTP logs

I was wondering what that ID was, it really confused me. So which log would this be in? I ran the following and got nothing back.

`postal logs | grep KDYPAIWU`

**Reply by willpower232** [reply]
It is something like that, it is also possible that these logs have already been rotated out of existence so if you're able to reproduce the issue, you can try something like `postal logs smtp` when you do so.

Alternatively you could be more direct with `docker compose logs smtp -f` or whatever to follow the logs as they happen.

**Reply by tdentwicklungsupport** [reply]
Hello @willpower232 

i like postal - thank you for your work.

Im currently in the process of upgrading/rebuilding or instance (from v1 to v3). We are facing the same? problem when using `smtp_relays` - they are just ignored and postal tries to send the mail using the MX record of the target domain.

Our line in `postal.yml`:
```
smtp_relays: ['smtp://10.242.142.53:25']
```

The only clue i found in the workerlogs ist this:
```
2024-07-22 06:48:29 +0000 ERROR  Cannot connect to <ip removed but not the one set in smtp_relays>:25 (<removed domain>) (Net::OpenTimeout: execution expired) log_id=PLFP1OKT component=worker thread=work1 original_queued_message=11 queued_message=11
```

We are using Postal only as a smtp server via HTTP :)

**Reply by tdentwicklungsupport** [reply]
Nevermind - i found my initial problem: I did not have the `smtp_relays` config under the `postal` section....

But now i am also stuck at `no host to try`

**Reply by tdentwicklungsupport** [reply]
@adamcooke Hi mate! I did see that you did a lot of refactor for the smtp_sender class lately - maybe you have a clue? Thanks 🙏

**Reply by tdentwicklungsupport** [reply]
ok i found the problem - `smtp_relays` can not handle ip adresses - it has to be a fqdn/hostname otherwise nothing will work or be logged.

In Summary my two problems where:
- `smtp_relays` not under the `postal` section
- Used ip adress instead of fqdn/hostname

I will add a pull request to update the description of the config to make it clear that IP adresses currently will not work

**Comment by anshdoespython** [comment]
im facing same issue please help, i have also nested smtp_relays under postal: and getting same error
"No SMTP servers were available for gmail.com. No hosts to try.
Support Ref: 5VQVQJAN"
